src/game/objects/har.c

```diff
@@ -1250,12 +1284,30 @@ void har_collide_with_projectile(object *o_har, object *o_pjt) {
     har *other = object_get_userdata(
         game_state_find_object(o_har->gs, game_state_get_player(o_har->gs, abs(h->player_id - 1))->har_obj_id));
 
-    if(h->state == STATE_FALLEN || h->state == STATE_STANDING_UP || h->state == STATE_WALLDAMAGE || h->health <= 0 ||
-       h->state >= STATE_VICTORY) {
+    if(h->state == STATE_STANDING_UP || h->state == STATE_WALLDAMAGE || h->health <= 0 || h->state >= STATE_VICTORY) {
         // can't hit em while they're down, or done
         return;
     }
 
+    // rehit mode is off
+    if(!o_har->gs->match_settings.rehit && h->state == STATE_FALLEN) {
+        log_debug("REHIT is off");
+        return;
+    }
+
+    // rehit mode is on, but the opponent isn't airborne or stunned
+    if(o_har->gs->match_settings.rehit && h->state == STATE_FALLEN &&
+       (!object_is_airborne(o_har) || h->endurance <= 0)) {
+        log_debug("REHIT is not possible %d %f %f %f", object_is_airborne(o_har), o_har->pos.x, o_har->pos.y,
+                  h->endurance);
+        return;
+    }
+
+    bool rehit = false;
+    if(o_har->gs->match_settings.rehit && h->state == STATE_FALLEN && object_is_airborne(o_har) && h->endurance > 0) {
+        rehit = true;
+    }
```

Comment:
And here. Actually, should this be some sort of getter function ?

Reply:
And done